README.md

```diff
@@ -1,2 +1,119 @@
-# neo4j-http
-A simple ruby HTTP client for Neo4j
+# Neo4j::Http
+
+The `Neo4j::Http` gem provides `Neo4j::Http::Client` as a thin wrapper around the [Neo4j HTTP API](https://neo4j.com/docs/http-api/current/).
+
+## Installation
+
+Add this line to your application's Gemfile:
+
+```ruby
+gem 'neo4j-http'
+```
+
+And then execute:
+
+    $ bundle install
+
+Or install it yourself as:
+
+    $ gem install neo4j-http
+
+## Configuration
+
+The client is configured by default via a set of environment variables from `[Neo4j::Http::Configuration](https://github.com/doximity/neo4j-http/blob/master/lib/neo4j/http/configuration.rb)`:
+
+* `NEO4J_URL`  - The base URL to connect to Neo4j at
+* `NEO4J_USER` - The user name to use when authenticating to neo4j
+* `NEO4J_PASSWORD` - The password of the user to be used for authentication
+* `NEO4J_DATABASE` - The database name to be used when connectiong.  By default this will be nil and the path used for connecting to Neo4j wil be `/db/data/transaction/commit` to make it compliant with v3.5 of neo4j
+* `NEO4J_HTTP_USER_AGENT` - The user agent name provided in the request - defaults to `Ruby Neo4j Http Client`
+* `NEO4J_REQUEST_TIMEOUT_IN_SECONDS` - The number of seconds for the http request to time out if provided, defaults to nil
+
+These configuration values can also be set during intialization like:
+
+```
+Neo4j::Http.configure do |config|
+  config.request_timeout_in_secods = 42
+end
+```
+
+## Usage
+
+The core interface can be directly accessed on `Neo4::Http::Client`:
+
+```
+# Execute arbitrary cypher commands
```

Comment:
I would probably break these up into subheadings. Right now it renders as a big wall of text

Reply:
Also lets throw a hint to the highlighter e.g. ```ruby